api/org/commcare/api/json/JsonActionUtils.java

```diff
@@ -0,0 +1,262 @@
+        if(result == FormEntryController.ANSWER_REQUIRED_BUT_EMPTY) {
+            ret.put(ApiConstants.RESPONSE_STATUS_KEY,"error");
+            ret.put(ApiConstants.ERROR_TYPE_KEY, "required");
+        }
+        else if(result == FormEntryController.ANSWER_CONSTRAINT_VIOLATED){
+            ret.put(ApiConstants.RESPONSE_STATUS_KEY, "error");
+            ret.put(ApiConstants.ERROR_TYPE_KEY, "restraint");
+            ret.put(ApiConstants.ERROR_REASON_KEY, prompt.getConstraintText());
+        }
+        else if (result == FormEntryController.ANSWER_OK){
+            ret.put(ApiConstants.QUESTION_TREE_KEY, walkToJSON(model, controller));
+            ret.put(ApiConstants.RESPONSE_STATUS_KEY,"accepted");
+        }
+        return ret;
+    }
+
+    /**
+     * Answer the question, return the updated JSON representation of the question tree
+     *
+     * @param controller the FormEntryController under consideration
+     * @param model the FormEntryModel under consideration
+     * @param answer the answer to enter
+     * @param index the form index of the question to be answered
+     * @return The JSON representation of the updated question tree
+     */
+    public static JSONObject questionAnswerToJson(FormEntryController controller,
+                                                  FormEntryModel model, String answer, String index){
+
+        FormIndex formIndex = indexFromString(index, model.getForm());
+
+        FormEntryPrompt prompt = model.getQuestionPrompt(formIndex);
+        return questionAnswerToJson(controller, model, answer, prompt);
+    }
+
+    /**
+     * Return the IAnswerData version of the string data input
+     *
+     * @param formEntryPrompt the FormEntryPrompt for this question
+     * @param data the String answer
+     * @return the IAnswerData version of @data above
+     */
+    public static IAnswerData getAnswerData(FormEntryPrompt formEntryPrompt, String data){
+
+        if(formEntryPrompt.getDataType() == Constants.DATATYPE_CHOICE){
+            int index = Integer.parseInt(data);
+
+            SelectChoice ans = formEntryPrompt.getSelectChoices().get(index -1);
+
+            return new SelectOneData(ans.selection());
+        } else if(formEntryPrompt.getDataType() == Constants.DATATYPE_CHOICE_LIST){
+            String[] split = parseMultiSelectString(data);
+            Vector<Selection> ret = new Vector<>();
+            for (String s: split){
+                int index = Integer.parseInt(s);
+                Selection ans = formEntryPrompt.getSelectChoices().get(index -1).selection();
+                ret.add(ans);
+            }
+            return new SelectMultiData(ret);
+        } else if (formEntryPrompt.getDataType() == Constants.DATATYPE_GEOPOINT){
+            return AnswerDataFactory.template(formEntryPrompt.getControlType(), formEntryPrompt.getDataType()).cast(
+                    new UncastData(data.replace(",", " ").replace("[","").replace("]", "")));
```

Comment:
`data.replace(",", " ").replace("[","").replace("]", "")` can you somehow lift this out to an intermediate var or helper method so that you can provide a description as to why this string manipulation is needed?